src/es-module-shims.js

```diff
@@ -163,8 +166,19 @@ function resolveDeps (load, seen) {
     resolvedSource += source.slice(lastIndex);
   }
 
-  if (resolvedSource.indexOf('//# sourceURL=') === -1)
+  let hasSourceMappingURL = false
+  resolvedSource = resolvedSource.replace(/\/\/# sourceMappingURL=(.*)\s*$/, (match, url) => {
+    hasSourceMappingURL = true
+    return match.replace(url, new URL(url, load.r))
+  });
+  let hasSourceURL = false
+  resolvedSource = resolvedSource.replace(/\/\/# sourceURL=(.*)\s*$/, (match, url) => {
+    hasSourceURL = true
+    return match.replace(url, new URL(url, load.r))
+  });
```

Comment:
If we _really_ wanted to optimize this we could detect these in the Wasm lexer side of things. Would literally save 1ms at most though but could be a future option.

Reply:
yeah, but then it would make lexer do some work that's probably only needed for `es-module-shims`. Let's keep that option on the table for the future though.